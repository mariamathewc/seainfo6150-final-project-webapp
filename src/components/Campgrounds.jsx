import React from 'react';
import styles from "./Campgrounds.module.css";
import { NavLink } from 'react-router-dom';
import { useLocation } from "react-router";

const Campgrounds = (props) => {

	const location = useLocation();

	

		return (
			
			<div className={styles.container}>
				<div className={styles.text}>
					<h2>Choose your camping season </h2>

					<ul className={styles.listing}>
						<li className={styles.formating} >
							<NavLink className={styles.links} to={{
								pathname: '/spring',
								state: {
									campdetail: location.state.details,
									code:0
								}
							}} exact activeClassName='active'  > <b>Spring </b> </NavLink>
						</li>
						<li className={styles.formating}>
							<NavLink className={styles.links}  to={{
								pathname: '/summer',
								state: {
									campdetail: location.state.details,
									code: 1
								}
							}}  exact activeClassName='active'  > <b>Summer </b> </NavLink>
						</li>
						<li className={styles.formating}>
							<NavLink className={styles.links}  to={{
								pathname: '/autumn',
								state: {
									campdetail: location.state.details,
									code: 2
								}
							}}  exact activeClassName='active'  > <b>Autumn </b> </NavLink>
						</li>
						<li className={styles.formating}>
							<NavLink className={styles.links}  to={{
								pathname: '/winter',
								state: {
									campdetail: location.state.details,
									code: 3
								}
							}}  exact activeClassName='active'  > <b>Winter </b> </NavLink>
						</li>
						<li className={styles.formating}>
							<NavLink className={styles.links}  to={{
								pathname: '/snow',
								state: {
									campdetail: location.state.details,
									code: 4
								}
							}}  exact activeClassName='active'  > <b>Snow </b> </NavLink>
						</li>
					</ul>
					
				</div>
			</div>

		)
	
}

export default Campgrounds 