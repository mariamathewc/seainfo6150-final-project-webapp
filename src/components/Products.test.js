import React from "react";
import Campgrounds from "./Campgrounds.jsx";
import { render } from "@testing-library/react";
import routeData from 'react-router';

const mockLocation = {
	pathname: 'localhost:3000/products',

}


describe("Products tests", () => {
	it("renders correctly", () => {
		const { container } = jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
		expect(container).toMatchSnapshot();
	});
});

