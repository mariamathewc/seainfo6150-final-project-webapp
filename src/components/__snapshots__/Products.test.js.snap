// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products tests renders correctly 1`] = `undefined`;
